src/Ldap.php

```diff
@@ -834,9 +938,14 @@ public function bind($username = null, $password = null)
             ErrorHandler::start(E_WARNING);
             $bind = ldap_bind($this->resource, $username, $password);
             ErrorHandler::stop();
-            if ($bind) {
+
+            if ($bind !== false) {
                 $this->boundUser = $username;
                 return $this;
+            } else {
```

Comment:
Isn't this ```else``` obsolete? When ```$bind``` was not false we returned and so there's no need for an ```else```

Reply:
I looked at this again but it looks right to me -- if $bind _is_ false, the else provides an opportunity to try again if the user set a nonzero `reconnectAttempts`.

Reply:
Here as well: Consider this code:

```php
if ($bind !== false) {
    $this->boundUser = $username;
    return $this;
}

if ($this->shouldReconnect($this->resource)) {
    return $this;
}
``` 

It does the same, as it allows an opportunity to try again when ```$bind``` is ```false```. But without the ```else```…